use-cases/NIC-passthrough-and-Kata.md

```diff
@@ -0,0 +1,217 @@
+
+```bash
+$ sudo vi /usr/share/defaults/kata-containers/configuration.toml
+```
+a. Set hotplug_vfio_on_root_bus = true
+
+b. Make sure you are using the PC machine type by verifying machine_type = "pc"
+
+
+
+#### Install *lshw* and *pciutils* (Ubuntu)
+
+```bash
+$ sudo apt install lshw pciutils
+```
+
+
+#### Find PCI address for the interface you wish to pass to the container and place it in a variable
+
+Say I want to pass **enp0s20f1** to the container (see what interfaces you have by running `$ ip link` ). 
+
+Get address: 
+
+```bash
+$ NIC=<interface>
+```
+For example: 
+```bash
+$ NIC=enp0s20f1
+```
+Place in BDF:
+```bash
+$ BDF=$(sudo lshw -class network -businfo -numeric | grep ${NIC} | awk '{print $1;}' | cut -d@ -f2)
+```
+Test BDF:
+```bash
+$ echo $BDF
+```
+
+```
+Expected results (something like):
+
+0000:00:14.1
+```
+
+#### Load *vfio-pci*  kernel module
+
+```bash
+$ sudo modprobe -i vfio-pci
+```
+Test that the module loaded correctly: 
+
+```bash
+$ sudo lsmod | grep vfio
+```
+
+#### Unbind the Interface from its current driver. 
+
+```bash
+$ sudo echo $BDF | sudo tee /sys/bus/pci/devices/$BDF/driver/unbind
+```
+
+#### Find vendor & device ID
+
+```bash
+$ sudo lspci -n -s $BDF
+```
+
+```
+Expected results (something like):
+
+00:14.1 0200: 8086:1f40 (rev 03)
+```
+
+
+#### Bind the device to vfio-pci using the result of the previous step.
+
+```bash
+$ sudo echo 8086 1f40 | sudo tee /sys/bus/pci/drivers/vfio-pci/new_id
+```
+
+> **Note:** The ` xxxx xxxx ` format is of the correct form and not a mistake.
+
+#### Check what the new device name is, now under vfio
+
+```bash
+$ sudo ls /dev/vfio
+```
+```
+Expected results (something like):
+
+ 39  vfio
+```
+
+> **Note:** The requested outcome here is the newly created “39”.
+
+#### All set here. Time to move to the Container’s side of the deal.
+
+
+### Part 2 – The Container
+
+#### Create the container you wish to pass the NIC into
+
+```bash
+docker run -it --runtime=kata-runtime --name <choose a name> --device /dev/vfio/<device name from previous step> -v /dev:/dev <image> <command>
+```
+   For example:
+
+```bash
+docker run -it --runtime=kata-runtime --name vfio_con --device /dev/vfio/39 -v /dev:/dev ubuntu bash
+```
+If you aren’t sure what just happened back there, here’s a breakdown of the command:
+- `docker run` - creates and runs the container. 
+- ` -it ` - docker command flags for “interactive” and “tty”.
+- ` --runtime=kata-runtime ` - tells docker to run the container as a Kata Container (if you set kata-runtime as the default runtime, this option is redundant).
+- ` --name vfio_con ` – name the container. It can be any name you choose. 
+- ` --device /dev/vfio/39 ` – this one is the important part. With this command, you tell docker to pass the NIC to the container. 
+- `-v /dev:/dev `– here you tell docker to map the devices folder of the host to the devices folder of the container.
```

Comment:
I'd consider removing most of this explanation, keeping only details of `--device` and `-v`.